### ISSUE TITLE
Add property documentation syntax for Python, CoffeeScript, Lua, Clojure, Io

### ISSUE THREAD

**Issue by nwinter** [codecombat/codecombat#1237]
(See also #399)

Calling all Scribes! In the spell palette, we currently have documentation like this:

![screenshot 2014-06-26 17 08 52](https://cloud.githubusercontent.com/assets/99704/3406439/40b6e150-fd8f-11e3-9cda-55eba03f1cba.png)

But some of these things are written with JavaScript-specific syntax. We need to extend all the current Components' property documentation to handle the new syntax for Python, CoffeeScript, Lua, Clojure and Io.

There's a new process with this that just uses the level editor patch system, so you don't have to set up your dev environment or anything.

There are too many properties/languages to easily track who is working on what in this GitHub issue, so please use this Google Doc to mark when you are doing or have done things, since we can't merge patches instantly. Note that not all properties have any syntax differences between languages, so many fields can just be marked done with no changes.
### 0. Claim your target Component/Language [in the spreadsheet](https://docs.google.com/a/codecombat.com/spreadsheets/d/1j-jTeY5xEjszuNigLiptlvFflf23tr_4n3FJttHVOVg/edit#gid=0).

![screenshot 2014-06-26 17 37 06](https://cloud.githubusercontent.com/assets/99704/3406624/30f264fc-fd93-11e3-9627-fcf5820ee2a2.png)
### 1. Open [any level in the level editor](http://codecombat.com/editor/level/greed).

![screenshot 2014-06-26 10 54 29](https://cloud.githubusercontent.com/assets/99704/3402357/fd6fc8fa-fd5a-11e3-8ebf-d29750909c97.png)

_Different levels will have a few different Components needing documentation, but most are shared between most levels._
### 2. Click the Components tab.

![screenshot 2014-06-26 10 55 23](https://cloud.githubusercontent.com/assets/99704/3402361/12e59606-fd5b-11e3-98d4-533293351859.png)
### 3. Click on one of the Components.

![screenshot 2014-06-26 10 56 10](https://cloud.githubusercontent.com/assets/99704/3402369/2daa6408-fd5b-11e3-9880-14ddbb7b52ea.png)
### 4. Click the Component's Settings sub-tab.

![screenshot 2014-06-26 10 57 33](https://cloud.githubusercontent.com/assets/99704/3402383/5f05a166-fd5b-11e3-8ec5-9538ce36ee66.png)
### 5. Open the Property Documentation at the bottom.

![screenshot 2014-06-26 10 56 53](https://cloud.githubusercontent.com/assets/99704/3402377/4717bd50-fd5b-11e3-86d7-11d2c84232dd.png)
### 6. Translate the JavaScript documentation into your programming language.

Look at `programmable.Allied` for a good example. Switch each field that has JavaScript-specific syntax to a multiple-language field, preserving the JavaScript one in a new `javascript` field, and adding your target language as another field.

Here's an example of changing from single-language (JavaScript) to multi-language:

![screenshot 2014-06-26 11 02 55](https://cloud.githubusercontent.com/assets/99704/3402435/26f610c0-fd5c-11e3-9738-f6d0147468f8.png) ![screenshot 2014-06-26 11 04 14](https://cloud.githubusercontent.com/assets/99704/3402456/50dbda50-fd5c-11e3-975b-10216a50889f.png)
### 7. Click "Submit Patch" under the chevron for the component to submit a patch (upper right).

![screen shot 2014-08-11 at 4 52 07 pm](https://cloud.githubusercontent.com/assets/204723/3884626/8fefcfe2-21b2-11e4-8343-b2148d1c2b7a.png)
### 8. Mark your language/Component "done" in the spreadsheet.

Good work!

**Comment by reesylou** [comment]
I have added Lua to most of the Snippets, and am gradually working through the other items.  Is the spreadsheet still used/up to date?

**Comment by nwinter** [comment]
Thanks @reesylou! The spreadsheet isn't up-to-date, so the latest state of what has property documentation and snippets is just going to be the Components themselves in the level editor. Most of the important ones are present in my test level, https://direct.codecombat.com/editor/level/items.

**Comment by reesylou** [comment]
Since I am in checking them all, should those items already with Lua snippets using 'self' be changed to 'hero'?  eg actions.Acts, isReady snippet is self:isReady("${1:cleave}")

**Comment by nwinter** [comment]
Yes, and not just Lua, but the other languages, too. But it's not that big a deal, because we automatically convert to the new `hero` format on the front-end in absence of writing some code to do the data migration in the backend: https://github.com/codecombat/codecombat/blob/master/app/views/play/level/tome/editor/autocomplete.coffee#L285-L293

**Comment by reesylou** [comment]
Did a heap of component patches, but wasn't logged in... so all those seemingly random Lua additions and changing self to hero patches were me in case you wondered.
